packages/plugins/plugin-script/src/components/NotebookStack/NotebookStack.tsx

```diff
@@ -17,74 +32,170 @@ import { type TypescriptEditorProps } from '../TypescriptEditor';
     </Stack>
   );
 };
 
-// TODO(burdon): Support calling named deployed functions (as with sheet).
-// TODO(burdon): Different section types (value, query, expression, prompt).
 // TODO(burdon): Display errors.
-// TODO(burdon): Allow moving cursor between sections (CMD Up/Down).
+// TODO(burdon): Support calling named deployed functions (as with sheet).
+
+const editorStyles = 'p-2 pis-3';
 
 type NotebookSectionProps = {
   cell: Notebook.Cell;
+  space?: Space;
   graph?: ComputeGraph;
-  onCellInsert?: (after: string | undefined) => void;
+  onCellInsert?: (type: Notebook.CellType, after: string | undefined) => void;
   onCellDelete?: (id: string) => void;
+  onCellRun?: (id: string) => void;
 } & Pick<TypescriptEditorProps, 'env'>;
 
-const NotebookSection = ({ cell, graph, env, onCellInsert, onCellDelete }: NotebookSectionProps) => {
+const NotebookSection = ({ cell, space, graph, env, onCellInsert, onCellDelete, onCellRun }: NotebookSectionProps) => {
   const { t } = useTranslation(meta.id);
-  const extensions = useMemo(() => {
-    return cell.script.target
-      ? [createDataExtensions({ id: cell.id, text: createDocAccessor(cell.script.target, ['content']) })]
-      : [];
-  }, [cell.script.target]);
 
   const name = graph?.expressions.value[cell.id]?.name;
   const value = graph?.values.value[cell.id];
+  const view = cell.view?.target;
 
+  // TODO(burdon): Dragging preview.
   return (
-    <StackItem.Root role='section' item={cell}>
-      <StackItem.Heading classNames='attention-surface'>
-        <StackItem.HeadingStickyContent>
-          <DropdownMenu.Root>
-            <DropdownMenu.Trigger asChild>
-              <StackItem.SigilButton>
-                <Icon icon='ph--list--regular' size={5} />
-              </StackItem.SigilButton>
-            </DropdownMenu.Trigger>
-            <DropdownMenu.Portal>
-              <DropdownMenu.Content>
-                <DropdownMenu.Viewport>
-                  <DropdownMenu.Item onClick={() => onCellInsert?.(cell.id)}>
-                    {t('notebook cell insert label')}
-                  </DropdownMenu.Item>
-                  <DropdownMenu.Item onClick={() => onCellDelete?.(cell.id)}>
-                    {t('notebook cell delete label')}
-                  </DropdownMenu.Item>
-                </DropdownMenu.Viewport>
-                <DropdownMenu.Arrow />
-              </DropdownMenu.Content>
-            </DropdownMenu.Portal>
-          </DropdownMenu.Root>
-        </StackItem.HeadingStickyContent>
+    <StackItem.Root role='section' item={cell} draggable>
+      <StackItem.Heading classNames='bs-full justify-between attention-surface'>
+        <StackItem.DragHandle>
+          <div className='flex justify-center p-2'>
+            <Icon icon='ph--dots-six-vertical--regular' size={4} />
+          </div>
+        </StackItem.DragHandle>
+        <DropdownMenu.Root>
+          <DropdownMenu.Trigger asChild>
+            <div className='flex justify-center p-2'>
```

Comment:
Please use `IconButton` if the trigger has `asChild`, otherwise remove `asChild` and style the trigger so there is no interceding element.

Reply:
i don't really understand this. let's discuss.

Reply:
It's a very standard Radix pattern used throughout the repo, for reference have a look for any .Trigger subcomponent with asChild. We never need interceding divs in situations like this.